ui/src/preview/preview-confirm-revert-changes-dialog.tsx

```diff
@@ -0,0 +1,216 @@
+        case 'referenceLine':
+            return 'reference-line';
+        case 'locationTrack':
+            return 'location-track';
+        case 'switch':
+            return 'switch';
+        case 'kmPost':
+            return 'km-post';
+    }
+};
+
+const onlyDependencies = (changesBeingReverted: ChangesBeingReverted): PublishRequest => {
+    const allChanges = changesBeingReverted.changeIncludingDependencies;
+    const reqType = changesBeingReverted.requestedRevertChange.type;
+    const reqId = changesBeingReverted.requestedRevertChange.id;
+    return {
+        trackNumbers: allChanges.trackNumbers.filter(
+            (tn) => reqType != PreviewSelectType.trackNumber || tn !== reqId,
+        ),
+        kmPosts: allChanges.kmPosts,
+        referenceLines: allChanges.referenceLines.filter(
+            (rl) => reqType != PreviewSelectType.referenceLine || rl !== reqId,
+        ),
+        switches: allChanges.switches.filter(
+            (sw) => reqType != PreviewSelectType.switch || sw !== reqId,
+        ),
+        locationTracks: allChanges.locationTracks.filter(
+            (lt) => reqType != PreviewSelectType.locationTrack || lt !== reqId,
+        ),
+    };
+};
+
+const publicationRequestSize = (req: PublishRequest): number =>
+    req.switches.length +
+    req.trackNumbers.length +
+    req.locationTracks.length +
+    req.referenceLines.length +
+    req.kmPosts.length;
+
+const TrackNumberItem: React.FC<{ trackNumberId: LayoutTrackNumberId; changeTime: TimeStamp }> = (
+    props,
+) => {
+    const { t } = useTranslation();
+    const trackNumbers = useTrackNumbers('DRAFT', props.changeTime);
+    const trackNumber = trackNumbers && trackNumbers.find((tn) => tn.id === props.trackNumberId);
+    return trackNumber === undefined ? (
+        <li />
+    ) : (
+        <li>
+            {t('publish.revert-confirm.track-number')} {trackNumber.number}
+        </li>
+    );
+};
+
+const ReferenceLineItem: React.FC<{
+    referenceLineId: ReferenceLineId;
+    changeTimes: ChangeTimes;
+}> = (props) => {
+    const { t } = useTranslation();
+    const trackNumbers = useTrackNumbers('DRAFT', props.changeTimes.layoutTrackNumber);
+    const referenceLine = useReferenceLine(
+        props.referenceLineId,
+        'DRAFT',
+        props.changeTimes.layoutReferenceLine,
+    );
+    if (referenceLine === undefined || trackNumbers === undefined) {
+        return <li />;
+    }
+    const trackNumber = trackNumbers.find((tn) => tn.id === referenceLine.trackNumberId);
+    return trackNumber === undefined ? (
+        <li />
+    ) : (
+        <li>
+            {t('publish.revert-confirm.reference-line')} {trackNumber.number}
+        </li>
+    );
+};
+
+const LocationTrackItem: React.FC<{ locationTrackId: LocationTrackId; changeTime: TimeStamp }> = (
+    props,
+) => {
+    const { t } = useTranslation();
+    const locationTrack = useLocationTrack(props.locationTrackId, 'DRAFT', props.changeTime);
```

Comment:
Meillä pitäisi nykyään olla mahdollisuus tehdä massahakuja antamalla joukon ID:itä (esim. `location-track-api.ts` funktio `getLocationTracks()`. Tämä vaikuttaisi paikalta jossa niiden käyttäminen voisi olla ihan järkevä, niin ei tarvisi tehdä yksittäisiä hakuja. Näille massahauille ei tosin ole vielä fronttipään cachea, joten jos voidaan olla varmoja että nämä tulevat melkein aina kakusta, niin sitten tämä nykyinen on parempi.

Tämä sama pätenee myös vaihteille ja kilometripylväille alempana.